README.md

```diff
@@ -1,18 +1,54 @@
 # libyui-rest-api
 
-YaST UI rest api framework for integration testing.
+Libyui UI REST API framework for integration testing.
 Project started by @lslezak, with support of @cwh42 and @OleksandrOrlov.
 
 The solution allows to query the UI properties over HTTP using a REST API.
-This allows to automate UI interaction steps and avoid screen-based tools.
-API allows reading properties of the UI, so entered text can be validated.
+This allows to automate the UI interaction steps and avoid screen-based tools.
+The API allows reading properties of the UI, so the displayed values can be validated.
+It also allows interacting with the UI (clicking buttons, toggling
+check boxes, entering text,...).
 
-To start application with rest API enabled, use following commands:
-* `xdg-su -c 'Y2TEST=1 YUI_HTTP_PORT=9999 yast2 host'` for QT
-* `sudo Y2TEST=1 YUI_HTTP_PORT=9999 yast2 host` for ncurses.
+This repository contains the shared functionlity, you need to install
+the additional bindings for the specific UI frontends
+((libyui-ncurses-rest-api)[https://github.com/libyui/libyui-ncurses-rest-api] or
+or (libyui-qt-rest-api)[https://github.com/libyui/libyui-ncurses-rest-api]).
 
-After that, you can get documentation how to interact with UI by accessing
-`http://localhost:9999`.
+### Features
+
+- Optional plugins which extend the standard libyui library
+  - Less dependencies
+  - Can be installed only when needed
+- Can read the whole structure of the currently displayed dialog
+- Can query only the selected widgets
+- Allows sending the user input (clicking buttons, entering text,...)
+- Uses the standard HTTP protocol and the JSON data format
+  - Not bound to any specific programming language or framework
+  - Easy integration with any testing framework
+- Optional remote access (by default accessible only from the same machine)
+- Optional client authentication (HTTP Basic Auth)
+  - :warning: But without any encryption it is still sent in a clear text!
+- IPv6 support (quite interesting for testing virtual machines, the IPv6 link-local
+  address is based on the MAC address, you can easily get the IPv6 address for
+  your testing machine)
+
+### TODO
+
+- [ ] Properties of some widgets are still missing
+- [ ] Allow sending more user actions
+- [ ] Some widgets do not send notify events when changed via the API
+- [ ] SSL encryption/peer verification (needed for secure transferring of sensitive data
+      like passwords)
+- [ ] Allow connection via Unix domain sockets
+
+### Usage
+
+To start the application with rest API enabled, use the following commands:
+* `xdg-su -c 'YUI_HTTP_PORT=9999 yast2 host'` for QT
```

Comment:
s/QT/Qt/

The Trolls insist that QT is Apple Quick Time :smiley:

Reply:
I see, real Trolls... :stuck_out_tongue_winking_eye: